test/libsolidity/solidityExecutionFramework.h

```diff
@@ -170,6 +173,22 @@ class ExecutionFramework
 	{
 		return encodeArgs(u256(0x20), u256(_arg.size()), _arg);
 	}
+	static u256 fixed(rational const& _value, int _fixedBits)
+	{
+		rational maxValue = (_value < 0) ? 		
+			rational(bigint(1) << 255, 1):
+			rational((bigint(1) << 256) - 1, 1);
+		unsigned convertingBits = 0;
+		rational value = _value;
+		while (value * 0x100 <= maxValue && value.denominator() != 1 && convertingBits < 256)
```

Comment:
yup, as I suspected, it's not shifting the value over really so much as it is making the value larger. I tested it with this.

```
BOOST_AUTO_TEST_CASE(fixed_type_arguments)
{
	char const* sourceCode = R"(
		contract C {
			function f(fixed a, fixed b) returns (fixed, fixed) {
				return (a, b);
			}
		}
	)";
	compileAndRun(sourceCode, 0, "C");
	rational a = rational(dev::bigint(1),dev::bigint(3));
	rational b = rational(dev::bigint(1),dev::bigint(2));
	std::cout << b << endl;
	std::cout << fixed(b,128) << std::endl;
	std::cout << callContractFunction("f(fixed,fixed)", fixed(a,128), fixed(b,128)) << std::endl;
	BOOST_CHECK(callContractFunction("f(fixed,fixed)", fixed(a,128), fixed(b,128)) == encodeArgs(fixed(a,128), fixed(b,128)));
}
```

and this was the result:

```
vor0220@CaptainNumNutz:solidity/build ‹fixedDataType*›$ soltest -t SolidityEndToEndTest/fixed_type_arguments
Running 1 test case...
1/2
170141183460469231731687303715884105728
[]
/Users/vor0220/webthree-umbrella/solidity/test/libsolidity/SolidityEndToEndTest.cpp:6923: error: in "SolidityEndToEndTest/fixed_type_arguments": check callContractFunction("f(fixed,fixed)", fixed(a,128), fixed(b,128)) == encodeArgs(fixed(a,128), fixed(b,128)) has failed

*** 1 failure is detected in the test module "EthereumTests"
```

This should be a simply shifted over 128 but that doesn't seem to be what's going on here.